cmd/dep/ensure.go

```diff
@@ -362,3 +367,30 @@ func deduceConstraint(s string) gps.Constraint {
 	// TODO: if there is amgibuity here, then prompt the user?
 	return gps.NewVersion(s)
 }
+
+func checkErrors(m map[string]gps.PackageOrErr) error {
+	noGoErrors, pkgErrors := 0, 0
+	for _, poe := range m {
+		if poe.Err != nil {
+			switch poe.Err.(type) {
+			case *build.NoGoError:
+				noGoErrors++
+			default:
+				pkgErrors++
+			}
+		}
+	}
+	if len(m) == 0 || len(m) == noGoErrors {
+		return errors.New("all dirs lacked any go code")
+	}
+
+	if len(m) == pkgErrors {
+		return fmt.Errorf("all dirs had go code with errors")
```

Comment:
No need for `Errorf()` when there's no variables being interpolated happening. Just `errors.New()` is fine